### PR TITLE
Export dbus address conditionally

### PR DESCRIPTION
**Pull request by keszybz**
Fixup for #11327. This takes the safer (I hope) route of reverting to the previous code.

### CODE REVIEW

File: src/login/pam_systemd.c [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include "path-util.h"
 #include "process-util.h"
 #include "socket-util.h"
+#include "string-util.h"
```
Comment (mrc0mmand) [inline src/login/pam_systemd.c]: I guess `#include "stdio-util.h"` should be here as well to fix the CentOS CI error

```
../src/login/pam_systemd.c: In function ‘pam_sm_open_session’:
../src/login/pam_systemd.c:433:17: error: implicit declaration of function ‘xsprintf’ [-Werror=implicit-function-declaration]
                 xsprintf(rt, "/run/user/"UID_FMT, pw->pw_uid);
```

**Comment by keszybz** [comment]
```
Jan 10 12:49:09 autopkgtest systemd-udevd[1313]: Failed to resolve group 'render': Connection timed out
Jan 10 12:49:34 autopkgtest systemd-udevd[1313]: Failed to resolve group 'render': Connection timed out
```
Same as all the other PRs filed today.

File: src/login/pam_systemd.c [hunk 2]
```diff
@@ -199,6 +199,18 @@ static int export_legacy_dbus_address(
         _cleanup_free_ char *s = NULL;
         int r = PAM_BUF_ERR;
 
+        /* FIXME: We *really* should move the access() check into the
+         * daemons that spawn dbus-daemon, instead of forcing
+         * DBUS_SESSION_BUS_ADDRESS= here. */
+
+        s = strjoin(runtime, "/bus");
```
Comment (poettering) [inline src/login/pam_systemd.c]: strjoina?

File: src/login/pam_systemd.c [hunk 3]
```diff
@@ -199,6 +199,18 @@ static int export_legacy_dbus_address(
         _cleanup_free_ char *s = NULL;
         int r = PAM_BUF_ERR;
 
+        /* FIXME: We *really* should move the access() check into the
+         * daemons that spawn dbus-daemon, instead of forcing
+         * DBUS_SESSION_BUS_ADDRESS= here. */
```
Comment (poettering) [inline src/login/pam_systemd.c]: i think we should conditionalize this as LEGACY or so? I don't think this is really worth supporting in new stuff

File: src/login/pam_systemd.c [hunk 4]
```diff
@@ -414,11 +415,10 @@ _public_ PAM_EXTERN int pam_sm_open_session(
 
         pam_get_item(handle, PAM_SERVICE, (const void**) &service);
         if (streq_ptr(service, "systemd-user")) {
-                _cleanup_free_ char *rt = NULL;
-
-                if (asprintf(&rt, "/run/user/"UID_FMT, pw->pw_uid) < 0)
-                        return PAM_BUF_ERR;
+                assert_cc(sizeof(uid_t) <= sizeof(int));
```
Comment (poettering) [inline src/login/pam_systemd.c]: why the assert? seems unnecessary to check this?

**Comment by poettering** [comment]
if we readd the access() check it really have a clear explanation what the usecase for this is in an adjacent comment

File: src/login/pam_systemd.c (hunk 3 above)
Comment (keszybz) [reply src/login/pam_systemd.c]: I prefer not to make things more complicated than they need to be. The tribulations we went through with systemd-240 suggest that dropping this is going to have bad effects even on recently written software. I think it's OK to export the variable without any plans to remove it.

**Pull request merged**